### PR TITLE
Fix `Validate::isArrayWithIds` method issue

### LINKED ISSUE #23791: Validate::isArrayWithIds method issue
## Describe the bug

The method `Validate::isArrayWithIds` return `true` when array is empty instead of  `false`.

## Expected behavior

The method should return `false` when `$ids` param is empty

## Steps to Reproduce

- Create `issue_23791.php` file in you root dir and put the following code 

```php

<?php 

    require(dirname(__FILE__).'/config/config.inc.php');
    $myEmptyArray = [];
    echo Validate::isArrayWithIds($myEmptyArray); // it will display 1
```
- Call `https://mytesstsite.com/issue_23791.php`

#### Additional information

* PrestaShop version: N/A
* PHP version: N/A

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | The method `Validate::isArrayWithIds` return `true` when array is empty instead of  `false`.
| Type?             | bug fix 
| Category?         | CO 
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | See https://github.com/PrestaShop/PrestaShop/issues/23791
| How to test?      | Unit test must be happy :)
| Possible impacts? | N/A.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23789)

### CODE REVIEW

File: classes/Validate.php [hunk 1]
```diff
@@ -1187,15 +1187,17 @@ public static function isLanguageFileName($file_name)
...
      */
     public static function isArrayWithIds($ids)
     {
-        if (count($ids)) {
-            foreach ($ids as $id) {
-                if ($id == 0 || !Validate::isUnsignedInt($id)) {
-                    return false;
-                }
+        if (count($ids) < 1) {
+            return false
```
Comment (kpodemski) [inline classes/Validate.php]:
```suggestion
            return false;
```

unit tests are not happy :trollface:

Comment (PululuK) [reply classes/Validate.php]: @kpodemski  It's Friday ! It's alcohol
Thanks man

File: classes/Validate.php [hunk 2]
```diff
@@ -1187,15 +1187,17 @@ public static function isLanguageFileName($file_name)
...
+     * @return bool return true if the array contain only unsigned int value and not empty
      */
     public static function isArrayWithIds($ids)
     {
-        if (count($ids)) {
-            foreach ($ids as $id) {
-                if ($id == 0 || !Validate::isUnsignedInt($id)) {
-                    return false;
-                }
+        if (count($ids) < 1) {
```
Comment (PierreRambaud) [inline classes/Validate.php]:
```suggestion
        if (!is_array($ids) || count($ids) < 1) {
```
Because the signature is not typed

Comment (PululuK) [reply classes/Validate.php]: @PierreRambaud  why `&&` ?

Comment (PierreRambaud) [reply classes/Validate.php]: Forget to valid my edit :sweat_smile:

File: tests-legacy/Unit/Classes/ValidateCoreTest.php [hunk 3]
```diff
@@ -365,4 +373,23 @@ public function isFloatDataProvider()
             )
         );
     }
+
+    public function isArrayWithIdsDataProvider()
+    {
+        return array(
+            array(true, [666]),
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php]:
```suggestion
            array(false, 'This is not an array'),
            array(true, [666]),
```

File: tests-legacy/Unit/Classes/ValidateCoreTest.php [hunk 4]
```diff
@@ -365,4 +373,24 @@ public function isFloatDataProvider()
             )
         );
     }
+
+    public function isArrayWithIdsDataProvider()
+    {
+        return array(
+            array(false, 'This is not an array'),
```
Comment (jolelievre) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php]:
```suggestion
            array(false, 'This is not an array'),
            array(false, 42),
            array(false, '42'),
```
Then these ones as well I guess?

**Review by kpodemski — DISMISSED** [review]

File: tests-legacy/Unit/Classes/ValidateCoreTest.php [hunk 5]
```diff
@@ -365,4 +373,26 @@ public function isFloatDataProvider()
             )
         );
     }
+
+    public function isArrayWithIdsDataProvider()
```
Comment (Progi1984) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php]:
```suggestion
    public function isArrayWithIdsDataProvider(): array
```

File: tests-legacy/Unit/Classes/ValidateCoreTest.php [hunk 6]
```diff
@@ -132,6 +132,14 @@ public function testIsOptFloat($expected, $input)
         $this->assertSame($expected, Validate::isOptFloat($input));
     }
 
+    /**
+     * @dataProvider isArrayWithIdsDataProvider
+     */
+    public function testIsArrayWithIds($expected, $input)
```
Comment (Progi1984) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php:138]:
```suggestion
    public function testIsArrayWithIds(bool $expected, $input)
```

Comment (Progi1984) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php]:
```suggestion
    /**
     * @dataProvider isArrayWithIdsDataProvider
     *
     * @param bool $expected
     * @param string|int|array<string|int|bool|array> $input
     */
    public function testIsArrayWithIds(bool $expected, $input)
```

File: tests-legacy/Unit/Classes/ValidateCoreTest.php [hunk 7]
```diff
@@ -365,4 +373,26 @@ public function isFloatDataProvider()
             )
         );
     }
+
+    public function isArrayWithIdsDataProvider()
+    {
+        return array(
```
Comment (Progi1984) [inline tests-legacy/Unit/Classes/ValidateCoreTest.php]: Replace `array()` by `[]`

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: tests-legacy/Unit/Classes/ValidateCoreTest.php (hunk 7 above)
Comment (PululuK) [reply tests-legacy/Unit/Classes/ValidateCoreTest.php]: Yep !

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
I reset CI

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
I did not find any occurence of `isArrayWithIds` so no QA needed AFAIK. Thank you @PululuK

**Pull request merged**